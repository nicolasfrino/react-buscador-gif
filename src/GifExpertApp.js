import React, {useState} from 'react';
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';


export const GifExpertApp = ( {defaultCategories = []}) => {

    //const categories = ['One Punch','Samurai X','Dragon Ball'];
    const [categories, setCategories] = useState(defaultCategories );


    return (
        <>
            <h2>Buscador de gif </h2>
            <hr/>

            <AddCategory setCategories={ setCategories }/>

            <ol>
                {
                    categories.map((category) => (
                        //<li key={categoria}>{categoria}</li>
                            <GifGrid 
                                key={category} category={category}
                            />
                         )
                    )
                }
            </ol>


        </>
    );
}


export default GifExpertApp;