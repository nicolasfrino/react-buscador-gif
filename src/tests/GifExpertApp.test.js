import React from 'react';
import '@testing-library/jest-dom'
import { GifExpertApp } from '../GifExpertApp';
import { shallow } from 'enzyme';

describe('Pruebas en <GifExpertApp />',  () => {
   
    test ('GifExpertApp',  () => {

        const wrapper =  shallow ( <GifExpertApp /> );
        expect( wrapper ).toMatchSnapshot();
    })

    test ('debe mostar lista de categorias',  () => {

        const categories = ['Goku','Vegeta'];

        const wrapper =  shallow ( <GifExpertApp defaultCategories={categories}/> );
        expect( wrapper ).toMatchSnapshot();
        expect (wrapper.find('GifGrid').length).toBe( categories.length);
    })

})
