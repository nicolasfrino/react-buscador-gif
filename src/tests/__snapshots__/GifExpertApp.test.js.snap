// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> GifExpertApp 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> debe mostar lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="Goku"
      key="Goku"
    />
    <GifGrid
      category="Vegeta"
      key="Vegeta"
    />
  </ol>
</Fragment>
`;
